posthog/temporal/data_imports/pipelines/source/mypy_plugin.py

```diff
@@ -0,0 +1,44 @@
+import typing
+
+from mypy.nodes import ARG_POS, Argument, Var
+from mypy.plugin import ClassDefContext, Plugin
+from mypy.plugins.common import add_method_to_class
+from mypy.plugins.dataclasses import dataclass_class_maker_callback
+from mypy.types import AnyType, TypeOfAny, UnboundType
+
+
+def transform(ctx: ClassDefContext) -> None:
+    """Dataclass handling and adding special methods."""
+    dataclass_class_maker_callback(ctx)
+
+    self_type = ctx.api.named_type(ctx.cls.fullname)
+    any_type = AnyType(TypeOfAny.special_form)
+
+    str_type = ctx.api.named_type_or_none("builtins.str")
+    if str_type is None:
+        str_type = UnboundType("builtins.str")  # type: ignore
```

Comment:
Don't ask me about this, I copied it from mypy's codebase as this is how they get types for `builtints.str`.